backend/grant/admin/views.py

```diff
@@ -843,57 +843,85 @@ def sql_ms(where: str):
-            us.refund_address IS NOT NULL
-            '''
-        ))),
-        # here we don't care about current refund_address of user, just that there has been a refund_tx_id
-        'refunded': str(ex(sql_pc_p(
-            '''
-            pc.status = 'CONFIRMED' AND 
-            pc.staking = FALSE AND 
-            pc.refund_tx_id IS NOT NULL AND 
-            p.stage IN ('CANCELED', 'FAILED')
-            '''
-        ))),
-        # if there is no user, or the user hasn't any refund_address
-        'donations': str(ex(sql_pc_p(
-            '''
-            pc.status = 'CONFIRMED' AND 
-            pc.staking = FALSE AND 
-            pc.refund_tx_id IS NULL AND 
-            (pc.user_id IS NULL OR us.refund_address IS NULL) AND 
-            p.stage IN ('CANCELED', 'FAILED')
-            '''
-        ))),
-        'gross': str(ex(sql_pc_p("pc.status = 'CONFIRMED' AND pc.refund_tx_id IS NULL"))),
-    }
+    def gen_quarter_date_range(year, quarter):
+        if quarter == 1:
+            return f"{year}-1-1", f"{year}-3-31"
+        if quarter == 2:
+            return f"{year}-4-1", f"{year}-6-30"
+        if quarter == 3:
+            return f"{year}-7-1", f"{year}-9-30"
+        if quarter == 4:
+            return f"{year}-10-1", f"{year}-12-31"
+
+    # contributions = {
+    #     'total': str(ex(sql_pc("status = 'CONFIRMED' AND staking = FALSE"))),
+    #     'staking': str(ex(sql_pc("status = 'CONFIRMED' AND staking = TRUE"))),
+    #     'funding': str(ex(sql_pc_p("pc.status = 'CONFIRMED' AND pc.staking = FALSE AND p.stage = 'FUNDING_REQUIRED'"))),
+    #     'funded': str(
+    #         ex(sql_pc_p("pc.status = 'CONFIRMED' AND pc.staking = FALSE AND p.stage in ('WIP', 'COMPLETED')"))),
+    #     # should have a refund_address
+    #     'refunding': str(ex(sql_pc_p(
+    #         '''
+    #         pc.status = 'CONFIRMED' AND
+    #         pc.staking = FALSE AND
+    #         pc.refund_tx_id IS NULL AND
+    #         p.stage IN ('CANCELED', 'FAILED') AND
+    #         us.refund_address IS NOT NULL
+    #         '''
+    #     ))),
+    #     # here we don't care about current refund_address of user, just that there has been a refund_tx_id
+    #     'refunded': str(ex(sql_pc_p(
+    #         '''
+    #         pc.status = 'CONFIRMED' AND
+    #         pc.staking = FALSE AND
+    #         pc.refund_tx_id IS NOT NULL AND
+    #         p.stage IN ('CANCELED', 'FAILED')
+    #         '''
+    #     ))),
+    #     # if there is no user, or the user hasn't any refund_address
+    #     'donations': str(ex(sql_pc_p(
+    #         '''
+    #         pc.status = 'CONFIRMED' AND
+    #         pc.staking = FALSE AND
+    #         pc.refund_tx_id IS NULL AND
+    #         (pc.user_id IS NULL OR us.refund_address IS NULL) AND
+    #         p.stage IN ('CANCELED', 'FAILED')
+    #         '''
+    #     ))),
+    #     'gross': str(ex(sql_pc_p("pc.status = 'CONFIRMED' AND pc.refund_tx_id IS NULL"))),
+    # }
```

Comment:
Left `contributions` commented because it's a great reference, but I can remove before merge if desired.